webapp/channels/src/components/preparing_workspace/organization.tsx

```diff
@@ -0,0 +1,207 @@
+        if (!inputRef.current?.value) {
+            return;
+        }
+        const name = inputRef.current?.value.trim();
+
+        const currentTeam = teams[0];
+
+        if (currentTeam && name && name !== currentTeam.display_name) {
+            const {error} = await props.updateTeam({...currentTeam, display_name: name});
+            if (error !== null) {
+                setApiCallError();
+            }
+        }
+    };
+
+    const createTeamFromOrgName = async () => {
+        if (!inputRef.current?.value) {
+            return;
+        }
+        const name = inputRef.current?.value.trim();
+
+        if (name) {
+            const {error, newTeam} = await props.createTeam(name);
+            if (error !== null || newTeam === null) {
+                props.setInviteId('');
+                setApiCallError();
+                return;
+            }
+            props.setInviteId(newTeam.invite_id);
+        }
+    };
+
+    const handleOnChange = (e: ChangeEvent<HTMLInputElement>) => {
+        props.setOrganization(e.target.value);
+        teamApiError.current = null;
+    };
+
+    const onNext = (e?: React.KeyboardEvent | React.MouseEvent) => {
+        if (e && (e as React.KeyboardEvent).key) {
+            if ((e as React.KeyboardEvent).key !== Constants.KeyCodes.ENTER[0]) {
+                return;
+            }
+        }
+        if (!triedNext) {
+            setTriedNext(true);
+        }
+
+        // if there is already a team, maybe because a page reload, then just update the teamname
+        const thereIsAlreadyATeam = teams.length > 0;
+        teamApiError.current = null;
+
+        if (!validation.error && !thereIsAlreadyATeam && props.isSelfHosted) {
+            createTeamFromOrgName();
+        } else if (!validation.error && thereIsAlreadyATeam && props.isSelfHosted) {
+            updateTeamNameFromOrgName();
+        }
+
+        if (validation.error || teamApiError.current) {
+            reportValidationError();
+            return;
+        }
+        props.next?.();
+    };
+
+    let className = 'Organization-body';
+    if (props.className) {
+        className += ' ' + props.className;
+    }
+    return (
+        <CSSTransition
+            in={props.show}
+            timeout={Animations.PAGE_SLIDE}
+            classNames={mapAnimationReasonToClass('Organization', props.transitionDirection)}
+            mountOnEnter={true}
+            unmountOnExit={true}
+        >
+            <div className={className}>
+                <div className='Organization-right-col'>
+                    <div className='Organization-form-wrapper'>
+                        <div className='Organization__progress-path'>
+                            <OrganizationSVG/>
+                            <PageLine
+                                style={{
+                                    marginTop: '5px',
+                                    height: 'calc(50vh)',
+                                }}
```

Comment:
Should this be moved into the scss?

Reply:
since the PageLine component would need these values to be different depending on where it is used, this act more as config params than as css values. Moving it to css would mean having to have different options for each scenario and passing to PageLine component a param which will be used to determine which css style classname with what marginTop and heigh to use.

Reply:
Thanks for that context!